docs/src/tutorials/yocto-linux.md

```diff
@@ -0,0 +1,102 @@
+# Build Thin Edge for a Yocto Linux distribution
+
+Yocto Project enables you to create a customised Linux distribution for your IoT devices. You can select the base image
+and add layers, containing software that you need on your image. In this tutorial, we will add Thin Edge using the
+`meta-tedge` layer. For more information, see the [getting started document on Yocto Project
+website](https://www.yoctoproject.org/software-overview/).
+
+
+`meta-tedge` is supported for **Yocto version 4.0 "Kirkstone"**. Additionally, we will depend `meta-openembedded` layer
+for Mosquitto.
+
+## Installation
+
+If you are not familiar with building Yocto distribution or you have not configured your build host yet, we strongly
+recommend to look into [official yocto documentation](https://docs.yoctoproject.org/brief-yoctoprojectqs/index.html)
+as the installation process will now skip all information that were mentioned there! For workspace organization or
+raspberry pi distribution, we also recommend this [guide](https://github.com/jynik/ready-set-yocto)
+
+> Most of the installation process is based on [Yocto Project Quick Build
+guide](https://docs.yoctoproject.org/brief-yoctoprojectqs/index.html).
+
+Before starting, be sure your hosts meets the following requirements:
+
+- 50 Gbytes of free disk space
+of Linux distributions that support the Yocto Project, see the Supported Linux Distributions section in the Yocto
+Project Reference Manual. For detailed information on preparing your build host, see the Preparing the Build Host
+section in the Yocto Project Development Tasks Manual.
+- Git 1.8.3.1 or greater
+- tar 1.28 or greater
+- Python 3.6.0 or greater.
+- gcc 5.0 or greater.
+
+For the purposes of the tutorial, we assume `~/yocto-thinedge` working directory. If using other directory, be sure to
+change paths where needed.
+
+### Build Host Packages
+
+Install essential packages:
+
+```bash
+$ sudo apt install gawk wget git diffstat unzip texinfo gcc build-essential chrpath socat cpio python3 python3-pip python3-pexpect xz-utils debianutils iputils-ping python3-git python3-jinja2 libegl1-mesa libsdl1.2-dev pylint3 xterm python3-subunit mesa-common-dev zstd liblz4-tool
+```
+
+### Clone Poky Linux repository
+
+Clone the Poky Linux distribution sources. We'll be using version `kirkstone`. You can use `--depth=1` to speed up the
+process. If using these options, to see previous commits or other branches see
+[here](https://stackoverflow.com/questions/29270058/how-to-fetch-all-git-history-after-i-clone-the-repo-with-depth-1)
+and [here](https://stackoverflow.com/questions/17714159/how-do-i-undo-a-single-branch-clone).
+
+```
+$ git clone git://git.yoctoproject.org/poky --branch=kirkstone --depth=1
+```
+
+### Add layers
+
+We'll be using `meta-tedge` and `meta-openembedded`. First, fetch the repositories:
+
+```bash
+git clone https://github.com/Bravo555/meta-tedge.git
```

Comment:
Since the project is outside of thin-edge, I wonder if the doc should be here or there.

Reply:
Okay, I have a better understanding now. It's okay to have this in an external repository.

However, I noticed that the recipes are specifically named after a release of thin-edge as for [tedge-agent_0.7.4.bb](https://github.com/Bravo555/meta-tedge/blob/main/recipes-tedge/tedge-agent/tedge-agent_0.7.4.bb). Does this mean that these recipes have to be re-written for each an all versions?

Reply:
> However, I noticed that the recipes are specifically named after a specific version of thin-edge as for [tedge-agent_0.7.4.bb](https://github.com/Bravo555/meta-tedge/blob/main/recipes-tedge/tedge-agent/tedge-agent_0.7.4.bb). Does this mean that these recipes have to be re-written for each and all versions?

Is this what you want to fix with https://github.com/thin-edge/thin-edge.io/issues/1343 ?

Reply:
> > However, I noticed that the recipes are specifically named after a specific version of thin-edge as for [tedge-agent_0.7.4.bb](https://github.com/Bravo555/meta-tedge/blob/main/recipes-tedge/tedge-agent/tedge-agent_0.7.4.bb). Does this mean that these recipes have to be re-written for each and all versions?
> 
> Is this what you want to fix with #1343 ?

Kind of. The recipes don't have to be re-written from scratch, as we generate them with `cargo-bitbake` tool, and what changes between each version is the revision of the source code we pull, and the list of dependencies. We have also created [a script](https://github.com/Bravo555/meta-tedge/blob/main/update-layer.sh) that should automate updating the recipes for all components, but for now it may not be completely foolproof and its result has to be carefully and manually inspected.

As for the configuration scripts, we have managed to make use of scripts which are in thin-edge.io source tree, in the `configuration` directory, so the scripts already present there might change and won't require any changes to the layer, but adding or removing scripts still requires manual intervention.

To be sure that the yocto layer is working correctly, we are working on automated tests that aim to verify that all system services and configuration scripts are working correctly.

Reply:
> To be sure that the yocto layer is working correctly, we are working on automated tests that aim to verify that all system services and configuration scripts are working correctly.

For that you can work with @gligorisaev.

Reply:
I think if we change repo to [meta-thin-edge](https://github.com/thin-edge/meta-thin-edge), we should also change it in documentation.

Reply:
> I think if we change repo to [meta-thin-edge](https://github.com/thin-edge/meta-thin-edge), we should also change it in documentation.

Yes, definitely. We also have to populate that repo before merging this PR.

Reply:
https://github.com/thin-edge/meta-tedge